src/traces/pie/layout_defaults.js

```diff
@@ -16,7 +16,9 @@ module.exports = function supplyLayoutDefaults(layoutIn, layoutOut) {
     function coerce(attr, dflt) {
         return Lib.coerce(layoutIn, layoutOut, layoutAttributes, attr, dflt);
     }
-    coerce('hiddenlabels');
+
     coerce('piecolorway', layoutOut.colorway);
     coerce('extendpiecolors');
+
+    if(!layoutOut.hasOwnProperty('hiddenlabels')) coerce('hiddenlabels');
```

Comment:
Why do we need this extra `if` clause?

Reply:
To coerce once.

Reply:
Are there any side-effects from coercing `hiddelabels` twice? If not, it's not worth adding complexity to skip one `coerce` call.

Reply:
Done in https://github.com/plotly/plotly.js/pull/3876/commits/abc3b39d159e5d4f53dc170b1d11f4fc3bc5f535.